plugins/basic/python/tk_maya_basic/plugin_engine.py

```diff
@@ -83,12 +83,14 @@ def bootstrap(sg_user, progress_callback, completed_callback, failed_callback):
 
     logger.debug("Will launch the engine with entity: %s" % entity)
 
-    # Remove the custom logging handler now that the engine will take over logging.
-    sgtk.LogManager().root_logger.removeHandler(plugin_logging_handler)
-
     # Install the bootstrap progress reporting callback.
     toolkit_mgr.progress_callback = progress_callback
 
+    # install a callback that turns off logging just before the
+    # engine starts up its own logging
+    callback = lambda ctx, log_handler=plugin_logging_handler: _pre_engine_start_callback(ctx, log_handler)
```

Comment:
Why are we even removing the plugin logging handler? Why can't it be the responsibility of the pre bootstrap phase to ensure that our global logger is hooked up to the DCCs for the duration of the session? This would simplify the whole thing and would guarantee you'd never miss a single message.